lang/py/src/avro/schema.py

```diff
@@ -722,14 +818,27 @@ def make_avsc_object(json_data, names=None):
   if hasattr(json_data, 'get') and callable(json_data.get):
     type = json_data.get('type')
     other_props = get_other_props(json_data, SCHEMA_RESERVED_PROPS)
+    logical_type = None
+    if 'logicalType' in json_data:
+      logical_type = json_data.get('logicalType')
+    if logical_type and logical_type != 'decimal':
+      raise SchemaParseException("Currently does not support %s logical type" % logical_type)
+    else:
+      precision = json_data.get('precision')
+      scale = 0 if json_data.get('scale') is None else json_data.get('scale')
     if type in PRIMITIVE_TYPES:
+      if type == 'bytes' and logical_type == 'decimal':
+        return DecimalSchema(type, precision, scale)
```

Comment:
Yes, I didn't like it much either. Nested `if else` always smells but it was written this way. I think if it needs to change then we would end up modifying a lot more which would be out of the scope of our ticket.